conf/modules.config

```diff
@@ -17,173 +17,91 @@ process {
-    }
-    withName: 'FASTK_MERGE' {
-        scratch    = false  // Disable scratch to get around bug https://github.com/nextflow-io/nextflow/issues/2983
-        ext.prefix = { "${meta.id}_merged" }
-    }
-    withName: '.*:BUILD_HIFI_FASTK_DATABASE:FASTK_FASTK' {
-        ext.prefix = { "${meta.id}_${reads[0].baseName}_hifi" }
-    }
-    withName: '.*:BUILD_HIC_FASTK_DATABASE:FASTK_FASTK' {
-        ext.prefix = { "${meta.id}_${reads[0].baseName}_hic" }
-    }
-    if( params.enable_ont_kmer_analyses ){
-        withName: '.*:BUILD_ONT_FASTK_DATABASE:FASTK_FASTK' {
-            ext.prefix = { "${meta.id}_${reads[0].baseName}_ont" }
-        }
-    }
-    withName: '.*:BUILD_ILLUMINA_FASTK_DATABASE:FASTK_FASTK' {
-        ext.prefix = { "${meta.id}_${reads[0].baseName}_illumina" }
-    }
-
-    withName: 'MERYL_.*' {
-        ext.args = { "k=${meta.kmer_size?: params.kmer_size}" }
-    }
-
-    withName: 'NANOPLOT' {
-        publishDir = [
-            path: { "$params.outdir/data_properties/nanoplot" },
-            mode: params.publish_dir_mode,
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
-        ]
-    }
-    // GENOME PROPERTIES
-    withName: 'FASTK_HISTEX' {
-        ext.args   = "-G"
-        ext.prefix = { "${meta.id}_histex" }
-    }
-    withName: 'GENOMESCOPE2' {
-        ext.args   = { [
-            "-k=${meta.kmer_size?: params.kmer_size}",
-            "-p=${meta.ploidy?: params.ploidy}"
-        ].join(' ') }
-        // Can move to subdir based on task.process
-        publishDir = [
-            path: { "$params.outdir/genome_properties/genomescope2" },
-            mode: params.publish_dir_mode,
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+    /* 
+    ----------
+    ASSEMBLY
+    ----------
+    */
+    // FLYE: Here I can switch to nf-core/flye
+    withName: FLYE {
+        ext.ags = [
+            "--genome-size ${params.genome_size}",
+            params.flye_args
         ]
-    }
-    withName: 'GENESCOPEFK' {
-        ext.args   = { [
-            "--kmer_length ${meta.kmer_size?: params.kmer_size}",
-            "--ploidy ${meta.ploidy?: params.ploidy}"
-        ].join(' ') }
-        publishDir = [
-            path: { "$params.outdir/genome_properties/genescopefk" },
-            mode: params.publish_dir_mode,
-            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        .join(" ")
+        .trim(  )
+    publishDir = [
+        path: { "${params.outdir}/assembly/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
         ]
     }
-    withName: 'MERQURYFK_PLOIDYPLOT' {
+    // HIFIASM:
+    /* Don't think I should switch to nf-core 
+       here since the UL process takes more inputs */
```

Comment:
Some inputs are optional and using [] will omit them, depending on the module. Its also possible to use the nf-core module and modify it slightly and then [patch the module](https://nf-co.re/docs/nf-core-tools/modules/patch). Having said that, local modules are imho also fine.